packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1133,53 +1133,62 @@ class ThemeData with Diagnosticable {
 
   /// A temporary flag used to opt-in to Material 3 features.
   ///
-  /// If true, then components that have been migrated to Material 3 will
+  /// If true, then widgets that have been migrated to Material 3 will
   /// use new colors, typography and other features of Material 3.
   /// If false, they will use the Material 2 look and feel.
   ///
+  /// During the migration to Material 3, turning this on may yield
+  /// inconsistent look and feel in your app as some widgets are migrated
+  /// while others have yet to be.
+  ///
+  /// Defaults to false. When the Material 3 specification is deemed complete
+  /// and all widgets are migrated on stable, we will change this flag to be
+  /// true by default. After that change has landed on stable, we will deprecate
+  /// this flag and remove all uses of it. At that point, the `material` library
+  /// will aim to only support Material 3.
+  ///
+  /// This flag affects styles and components.
+  ///
+  /// ## Styles
+  ///   * [Color](https://m3.material.io/styles/color/the-color-system): [ColorScheme], [Material]
+  ///   * [Shape](https://m3.material.io/styles/shape) (see components below)
+  ///   * [Typography](https://m3.material.io/styles/typography): `typography` (see table below)
+  ///
+  /// ## Components
+  ///   * [Common buttons](https://m3.material.io/components/buttons): [TextButton], [OutlinedButton], [ElevatedButton]
+  ///   * [FAB](https://m3.material.io/components/floating-action-button): [FloatingActionButton]
+  ///   * [Extended FAB](https://m3.material.io/components/extended-fab): [FloatingActionButton.extended]
+  ///   * [Cards](https://m3.material.io/components/cards): [Card]
+  ///   * [Dialogs](https://m3.material.io/components/dialogs): [Dialog], [AlertDialog]
+  ///   * Lists: [ListTile]
+  ///   * [Navigation bar](https://m3.material.io/components/navigation-bar): [NavigationBar] (new, replacing [BottomNavigationBar])
+  ///   * [Navigation rail](https://m3.material.io/components/navigation-rail): [NavigationRail]
+  ///   * [Top app bar](https://m3.material.io/components/top-app-bar): [AppBar]
+  ///
+  ///
+  /// In addition, this flag enables features introduced in Android 12.
+  ///   * Stretch overscroll: [MaterialScrollBehavior]
+  ///   * Ripple: `splashFactory` (see table below)
+  ///
+  /// ## Defaults
   /// If a [ThemeData] is constructed with [useMaterial3] set to true, then
-  /// some properties will get special defaults. However, just copying a [ThemeData]
-  /// with [useMaterial3] set to true will not change any of these properties in the
-  /// resulting [ThemeData]. These properties are:
```

Comment:
I think we need this description (or one like it here), as this is a distinction that a lot of people will miss and be confused when they don't get all the M3 features. A lot of apps tend to create a ThemeData with some default constructor like `ThemeData.light()` and then use `.copyWith` to configure it. If they do that with this property they won't get the new typography or splash factory defaults.

I would also like to see this above the component list as this is easily missed here at the bottom of the doc.

Reply:
Hmm I think that's a bug then. I would expect `copyWith` with `useMaterial3` true to apply these updates, if I haven't provided my own `typography` and `splashFactory`, of course. 

If I understand correctly, we're currently requiring *every created theme out there* (that doesn't use the factory constructor) to add the following?

```.copyWith(
...
typography: Typography.material2021(platform: platform),
splashFactory: platform == TargetPlatform.android && !kIsWebInkSparkle.splashFactory ?  InkRipple.splashFactory
      : InkSplash.splashFactory,
)
```

It's not documented anywhere and is pretty brittle.

Nevertheless, that's out of scope for this PR, so I've moved the section above the components section

Reply:
> Hmm I think that's a bug then. I would expect `copyWith` with `useMaterial3` true to apply these updates, if I haven't provided my own `typography` and `splashFactory`, of course.

Not a bug so much as an easily misunderstood feature. `copyWith` just blindly overrides any given parameters, but does no other processing. Even if we want to special case this for `useMaterial3`, once a ThemeData is constructed, it will have non-null typography and splashFactory settings. We would not be able to tell if you have provided your own or if they were default.

> If I understand correctly, we're currently requiring _every created theme out there_ (that doesn't use the factory constructor) to add the following?
> ```
> ...
> typography: Typography.material2021(platform: platform),
> splashFactory: platform == TargetPlatform.android && !kIsWeb ? InkSparkle.splashFactory ?  InkRipple.splashFactory,
> )
> ```

There is no way to construct a ThemeData that doesn't use the factory constructor other than `ThemeData.raw` or `ThemeData.copyWith` (which itself uses `.raw`). Unless I am missing something here.

If people want to use `copyWith` in conjunction with `useMaterial3` they will get most of the features, just not the ones that have defaults set in the factory constructor (hence the description in the docs).

> Nevertheless, that's out of scope for this PR, so I've moved this section above the styles and components sections.

Thanks.